### PR TITLE
Name columns

### PR DESCRIPTION
**Pull request by megan-godfrey**
Column orders...
When not verbose and output to TTY: Index, ID, Name
When verbose and output to TTY: Index, ID, Name, File
When verbose and not output to TTY: Index, ID, File, Name

### CODE REVIEW

**Review by benarmston — CHANGES_REQUESTED** [review]
My first pass through this and it looks about right.  I've added a comment about indentation and vim settings.  Could you take a look at that please and I'll take another look at this afterwards.

**Review by benarmston — CHANGES_REQUESTED** [review]
Thanks for fixing the indentation @megan-godfrey.  Could you take a look at the comment below.  If any of it doesn't make sense, let me know.

File: lib/flight_job/outputs/list_templates.rb [hunk 1]
```diff
@@ -62,6 +69,14 @@ def register_all
           end
         end
       end
+            
+      if $stdout.tty?
```
[the remark this answers is not in the document]
Comment (benarmston) [reply lib/flight_job/outputs/list_templates.rb:73]: In the final example above the "File" and "Name" columns are also the wrong way around.  We want

```
/opt/flight/bin/bundle exec ./bin/job list-templates --ascii --verbose
```

and

```
/opt/flight/bin/bundle exec ./bin/job list-templates --ascii --verbose | cat
```
to produce the same output.

**Review by benarmston — CHANGES_REQUESTED** [review]
Thanks for making those changes @megan-godfrey.

This should be the final set of changes I'm asking you to make on this PR.

The `register_all` method has become busier than I'd normally like.  It's currently doing two things which are probably better separated.  The first of these is defining which columns are registered in which order.  The second is defining what the column consists of; its header text, colour and content.

The `register_all` method would be better as (something like):

```ruby
def register_all
  register_index
  register_id
  if humanize? 
    register_name
    register_file if verbose?
  else
    register_file
    register_name
  end
end
```

I think this implementation makes it easier to ensure that the order of the columns for the machine-readable output remains consistent.

The new methods e.g., `register_index`, can then be implemented as `private` methods.

Does what I'm after and why I'm after it make sense?

**Comment by megan-godfrey** [comment]
That makes sense, yeah. OK I'll get working on that :)

**Comment by benarmston** [comment]
Thanks for you work on this @megan-godfrey .  LGTM.

**Pull request merged**